js/donations-visualiser.js

```diff
@@ -887,6 +887,12 @@ function processData(data) {
     filterData();
 }
 
+
+function toTrainCase(str) {
+    if (!str || typeof str !== 'string') return str;
+    return str.toLowerCase().replace(/ /g, '-');
```

Comment:
should this be:
str.toLowerCase().replace(/\s\+/g, '-');

or do you want "hello  ptim" to map to "hello--ptim"? the above version will also handle tabs in case that's useful.